lib/ssz_ex.ex

```diff
@@ -184,6 +192,185 @@ defmodule LambdaEthereumConsensus.SszEx do
     end
   end
 
+  defp encode_container(container, schemas) do
+    values =
+      schemas
+      |> Enum.map(fn {key, schema} ->
+        value = Map.get(container, key)
+        {value, schema}
+      end)
+
+    with {:ok,
```

Comment:
This is a very smart implementation! reduce_while is perfectly applied. Nevertheless, it's a bit convoluted with respect to code readability. I propose a refactor. We should be able to separate this into three parts:

```elixir
{fixed_size_values, fixed_length, variable_values} = analyze_schemas(schemas)
with {:ok, variable_parts} <- encode_schemas(variable_values),
     offsets = calculate_offsets(variable_parts, fixed_length),
     variable_length = Enum.reduce(variable_parts, 0, fn part, acc -> length(part) + acc end),
     :ok <- check_length(fixed_length + variable_length)
     {:ok, fixed_parts} <- replace_offsets(fixed_size_values, offsets) |> encode_schemas() do
       fixed_parts ++ variable_parts
end
```

This helper function might help:

```elixir
defp encode_schemas(schemas) do
  Enum.map(variable_values, fn {value, schema} -> encode(value, schema) end)
  |> flatten_results() 
end
```

The `analyze_schemas` function should return:

- `fixed_size_values`, which is a list of {value, schema} tuples for fixed size elements interleaved with `:offset` values.
- `fixed_length`, which is the sum of the fixed elements and the offset lengths. We can use the function `get_fixed_size` internally.
- `variable_values`m which is a list of {value, schema} only for variable sized elements.